digital_comms/runner.py

```diff
@@ -357,46 +489,49 @@ def run():
 
             total_distributions = [distribution for distribution in system._distributions]
 
-            adoption_desirability_percentage = (
+            adoption_desirability_percentage = round((
```

Comment:
I'm a bit nervous about the use of `round` within many of the calculations. These are likely to introduce potentially significant rounding errors.

Generally, you should only round when printing out results tables, and avoid at all costs inside the calculations are they are likely to introduce cumulative errors.